server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -607,6 +629,12 @@ public double bucketSize(Rounding.DateTimeUnit unit) {
         return innerSizedBucketAggregators[0].bucketSize(unit);
     }
 
+    @Override
+    public void collectDebugInfo(BiConsumer<String, Object> add) {
+        super.collectDebugInfo(add);
+        add.accept("num_docs_visited_no_index_sorting", numDocsVisitedNoIndexSorting);
+    }
```

Comment:
This adds an addition in a hot loop. Maybe a better debug information would be to tell the number of segments on which `init()` was called, which would tell us whether dynamic pruning kicked in or not.